### ISSUE TITLE
Matomo updates perfectly via http proxy - Marketplace does not work at all

### ISSUE THREAD

**Issue by cklabautermann** [matomo-org/matomo#15474]
Good morning,

I have this issue for a long time and would like to get rid of it.

My setup is behind a firewall. incoming tracking requests hit matomo via DNAT, so source ip´s stay untouched.
Outgoing requests like update checks are done via a intransparent proxy.
While matomo updates work perfectly via this setup, marketplace does not work at all.

a klick on "Marketplace" in the backend leads to the following line in apache error.log:

` [client <PROXY_IP>:40442]Error in Matomo: There was an error reading the response from the Marketplace: Please try again later.`

The proxy itself logs:

`srcip="<MATOMO_IP" dstip="185.31.40.177.....statuscode="200""`

http-proxy is defined in ../matomo/config/config.ini.php:

`[proxy]`
`host = "PROXY_IP"`
`port = PROXY_PORT`

Thanks for help and best regards,
Claudio

**Comment by tsteur** [comment]
By the looks both marketplace and core is using the same method etc to download the data. Is there any way you could get more information what is happening there or maybe even debug what is maybe happening in the `core/Http.php` file?

**Comment by cklabautermann** [comment]
@tsteur I was helped in the forum. They asked if the reason could be SSL related. I remembered that updating my Matomo only works via insecured http connection. 
I was able to change the proxy from intransparent to transparent and deleted the proxy lines in the config.ini.php. 
It was confusing to me that in the configuration of the marketplace plugin the URL is mentioned as 'http://plugins.matomo.org' and so I did not think that SSL (redirection) could be involved. 
Fortunately this was easily solved.
Thanks and regards,
Claudio
